### PR TITLE
[Stats] Add stats to dataset and generated train table

### PR DESCRIPTION
**Pull request by zechengz**
* Add stats to dataset and generated train table
* Move `examples/inferred_stypes.py` to each dataset
* Some code refactoring and cleaning

### CODE REVIEW

File: relbench/data/task_node.py [hunk 1]
```diff
@@ -93,3 +107,34 @@ def __init__(self, dataset: str, process: bool = False) -> None:
...
+            )
+            target_col_stats = compute_col_stats(
+                temp_df[self.target_col],
+                target_stype,
+            )
+            res[str(timestamp)] = {
+                self.entity_col: entity_col_stats,
+                self.target_col: target_col_stats,
+            }
+        return res
```
Comment (weihua916) [inline relbench/data/task_node.py:140]: what is it used for?

File: relbench/datasets/amazon.py [hunk 2]
```diff
@@ -252,3 +252,29 @@ def fix_column(value):
...
+            "review": {
+                "review_text": stype.text_embedded,
+                "summary": stype.text_embedded,
+                "review_time": stype.timestamp,
+                "rating": stype.numerical,
+                "verified": stype.categorical,
+                "customer_id": stype.numerical,
+                "product_id": stype.numerical,
+            },
+        }
```
Comment (weihua916) [inline relbench/datasets/amazon.py:280]: This should not an inherent property of a dataset. Rather, it's something related to example code that uses pytorch frame.

File: relbench/data/task_link.py [hunk 3]
```diff
@@ -148,3 +162,37 @@ def __init__(self, dataset: str, process: bool = False) -> None:
 
         def pack_tables(self, root: Union[str, os.PathLike]) -> Tuple[str, str]:
             return _pack_tables(self, root)
+
+    def stats(
```
Comment (zechengz) [inline relbench/data/task_link.py:166]: How many train table links (repeated purchases)

File: relbench/data/task_node.py [hunk 4]
```diff
@@ -93,3 +107,34 @@ def __init__(self, dataset: str, process: bool = False) -> None:
 
         def pack_tables(self, root: Union[str, os.PathLike]) -> Tuple[str, str]:
             return _pack_tables(self, root)
+
+    def stats(
```
Comment (zechengz) [inline relbench/data/task_node.py:111]: Number of trues and number of false for binary
regression: min, max and mean etc.

File: relbench/datasets/amazon.py (hunk 2 above)
Comment (zechengz) [reply relbench/datasets/amazon.py]: Moved back

File: relbench/data/task_node.py (hunk 1 above)
Comment (zechengz) [reply relbench/data/task_node.py]: This is the train / val / test table statistics, proposed by @rusty1s

**Review by XinweiHe — APPROVED** [review]

File: relbench/data/task_node.py (hunk 1 above)
Comment (weihua916) [reply relbench/data/task_node.py]: yes, but where are we using it?

Comment (zechengz) [reply relbench/data/task_node.py]: Seems to add this for the train table stats listed in the paper

Comment (zechengz) [reply relbench/data/task_node.py]: Discussed with @rusty1s , we are going to add the target distribution etc. to the appendix

**Review by weihua916 — APPROVED** [review]

File: relbench/data/task_node.py (hunk 1 above)
Comment (weihua916) [reply relbench/data/task_node.py]: Ok!

**Pull request merged**